### ISSUE TITLE
Video playback never loads with Amazon Fire TV

### ISSUE THREAD

**Issue by PrimordialHelios** [jellyfin/jellyfin-androidtv#82]
I recently set up a Jellyfin server and installed the app on my Fire TV Stick. I can connect just fine and play videos through the browser, and I can log in and browse through the app. However, playback won't work through the app. I select the video, and a blue spinner shows up and the media just never loads. I'm not sure if this is a bug or PEBKAC, but I'm not really sure where to start and I was hoping someone here could help me out.

Jellyfin app version: Beta 4 amazon release
Server version: 10.2.2
I use Apache as a reverse proxy and letsencrypt for TLS. I can also pastebin logs if needed.

**Comment by anthonylavado** [comment]
Specifically, did you install the Release or Debug version of the Amazon app?

**Comment by thornbill** [comment]
Also any information you could provide about the format of the videos you are trying to play would be helpful. I use Fire TV devices for testing and haven't encountered this issue yet.

**Comment by PrimordialHelios** [comment]
@anthonylavado release: `jellyfin-androidtv_amazon-release_beta4.apk`.

@thornbill avi and mkv

**Comment by PrimordialHelios** [comment]
I looked into the logs, and the last lines that have to do with the app are:

    [2019-04-05 21:09:28.259 -04:00] [INF] "HTTP POST" "http://jellyfin.primhelios.duckdns.org:8096/emby/Items/0858ef5e8dab3f082ebf75f23d059dc8/PlaybackInfo?format=json". UserAgent: "Dalvik/2.1.0 (Linux; U; Android 5.1.1; AFTT Build/LVY48F)"
    [2019-04-05 21:09:28.320 -04:00] [ERR] Error processing request
    System.ArgumentNullException: Value cannot be null.
    Parameter name: id
       at Emby.Server.Implementations.Library.UserManager.GetUserById(Guid id)
       at MediaBrowser.Api.Playback.MediaInfoService.GetPlaybackInfo(Guid id, Guid userId, String[] supportedLiveMediaTypes, String mediaSourceId, String liveStreamId)
       at MediaBrowser.Api.Playback.MediaInfoService.GetPlaybackInfo(GetPostedPlaybackInfo request)
       at MediaBrowser.Api.Playback.MediaInfoService.Post(GetPostedPlaybackInfo request)
       at Emby.Server.Implementations.Services.ServiceExecGeneral.GetTaskResult(Task task)
       at Emby.Server.Implementations.Services.ServiceHandler.ProcessRequestAsync(HttpListenerHost appHost, IRequest httpReq, IResponse httpRes, ILogger logger, String operationName, CancellationToken cancellationToken)
       at Emby.Server.Implementations.HttpServer.HttpListenerHost.RequestHandler(IHttpRequest httpReq, String urlString, String host, String localPath, CancellationToken cancellationToken)

Would this have anything to do with the issue?

**Comment by PrimordialHelios** [comment]
Curiouser and curiouser, beta-3 works perfectly fine. If I can get a toolchain set up, I'll try and find the commit that breaks for me.

**Comment by PrimordialHelios** [comment]
The issue might have something to do with the client app trying to use `http://mysite.com:8096` despite me entering `https://mysite.com` which is already set up to reverse-proxy to `localhost:8096`

Checking the logs, there isn't a single request to `https://mysite.com`. Is the client app sending my password over unencrypted http?

**Comment by thornbill** [comment]
That almost sounds like a dns issue. There is a known issue when entering the server url in the app if it does not contain a `:` then it defaults to using port `8096`.

If your site is using http then the password would be sent in cleartext.

**Comment by PrimordialHelios** [comment]
My site is not using http, it's using https, and I explicitly entered `https` when setting the url in the client.
Sending the password in cleartext without any sort of warning seems pretty bad.

**Comment by PrimordialHelios** [comment]
Removing the ternary expression in [this line](https://github.com/jellyfin/jellyfin-androidtv/blob/54ba26c58eb13001cc6bf86c4af41f650423c6b2/app/src/main/java/org/jellyfin/androidtv/util/apiclient/AuthenticationHelper.java#L55) so as to not add a port number makes the app work fine.

Perhaps that ternary should be removed, and the dialogue box should tell users how to specify the port with a colon.

**Comment by PrimordialHelios** [comment]
Alright

After a few hours, it looks like the issue is actually with the jellyfin server and some weirdness with ports. I've since forwarded 8096 and 8920, as well as added a certificate and required encrypted remote connections. After doing all that, the client now connects just fine over an encrypted connection.

Closing this since it seems like the client had absolutely nothing to do with it :P

Thanks for the help guys!

**Comment by thornbill** [comment]
> Sending the password in cleartext without any sort of warning seems pretty bad.

Well if you are using https everything using that connection will be encrypted. If you are using http then it’s not secure. Maybe we could display a warning in the app (most modern web browsers do) but I feel like that’s something the server administrator should already be aware of. 

> Also, manually specifying the port with https://mysite.com:80 doesn't work, the client refuses to connect.

Typically https uses port 443. You could also try with just `https://mysite.com`.

> Perhaps that ternary should be removed, and the dialogue box should tell users how to specify the port with a colon.

The existing implementation there is poor. At a minimum it should parse the url and explicitly check for a port number instead of just if the url contains a colon. I plan to update the login page and move the port input to a separate field like the web login currently uses.

**Comment by PrimordialHelios** [comment]
>but I feel like that’s something the server administrator should already be aware of.

That's the thing, I had SSL set up in my reverse proxy, and I specified that the client should use HTTPS. The client for some reason fell back to unencrypted HTTP without alerting me *at all*.

>Typically https uses port 443. You could also try with just `https://mysite.com`.

Thaaaaat would be my issue. Whoops, thanks. However, several times I specified `https://mysite.com` and the client *still* tried accessing the server at port 8096 over http.

My big issue here was that I specified that the connection should be encrypted, and the client attempted to use an unencrypted connection anyway.
